dlib/test/math.cpp

```diff
@@ -0,0 +1,497 @@
+            0.97974649, 0.98771489, 0.99371944, 0.99773596, 0.99974827,
+            0.99974827, 0.99773596, 0.99371944, 0.98771489, 0.97974649,
+            0.96984631, 0.95805423, 0.94441772, 0.92899171, 0.91183829,
+            0.89302655, 0.87263222, 0.85073744, 0.82743037, 0.80280484,
+            0.77696003, 0.75      , 0.72203331, 0.69317256, 0.66353398,
+            0.63323691, 0.60240333, 0.57115742, 0.53962498, 0.50793298,
+            0.47620904, 0.4445809 , 0.41317591, 0.38212053, 0.35153981,
+            0.32155689, 0.29229249, 0.26386446, 0.23638727, 0.20997155,
+            0.18472367, 0.16074529, 0.13813298, 0.11697778, 0.09736487,
+            0.07937323, 0.06307531, 0.04853673, 0.03581603, 0.02496444,
+            0.01602565, 0.00903565, 0.00402259, 0.00100666, 0.
+    };
+
+    /*! Data generated using scipy.signal.windows.hann(100, sym=False) !*/
+    const static double HANN_PERIODIC[100] = {
+            0.00000000e+00, 9.86635786e-04, 3.94264934e-03, 8.85637464e-03,
+            1.57084194e-02, 2.44717419e-02, 3.51117571e-02, 4.75864738e-02,
+            6.18466600e-02, 7.78360372e-02, 9.54915028e-02, 1.14743379e-01,
+            1.35515686e-01, 1.57726447e-01, 1.81288005e-01, 2.06107374e-01,
+            2.32086603e-01, 2.59123163e-01, 2.87110354e-01, 3.15937724e-01,
+            3.45491503e-01, 3.75655056e-01, 4.06309343e-01, 4.37333383e-01,
+            4.68604740e-01, 5.00000000e-01, 5.31395260e-01, 5.62666617e-01,
+            5.93690657e-01, 6.24344944e-01, 6.54508497e-01, 6.84062276e-01,
+            7.12889646e-01, 7.40876837e-01, 7.67913397e-01, 7.93892626e-01,
+            8.18711995e-01, 8.42273553e-01, 8.64484314e-01, 8.85256621e-01,
+            9.04508497e-01, 9.22163963e-01, 9.38153340e-01, 9.52413526e-01,
+            9.64888243e-01, 9.75528258e-01, 9.84291581e-01, 9.91143625e-01,
+            9.96057351e-01, 9.99013364e-01, 1.00000000e+00, 9.99013364e-01,
+            9.96057351e-01, 9.91143625e-01, 9.84291581e-01, 9.75528258e-01,
+            9.64888243e-01, 9.52413526e-01, 9.38153340e-01, 9.22163963e-01,
+            9.04508497e-01, 8.85256621e-01, 8.64484314e-01, 8.42273553e-01,
+            8.18711995e-01, 7.93892626e-01, 7.67913397e-01, 7.40876837e-01,
+            7.12889646e-01, 6.84062276e-01, 6.54508497e-01, 6.24344944e-01,
+            5.93690657e-01, 5.62666617e-01, 5.31395260e-01, 5.00000000e-01,
+            4.68604740e-01, 4.37333383e-01, 4.06309343e-01, 3.75655056e-01,
+            3.45491503e-01, 3.15937724e-01, 2.87110354e-01, 2.59123163e-01,
+            2.32086603e-01, 2.06107374e-01, 1.81288005e-01, 1.57726447e-01,
+            1.35515686e-01, 1.14743379e-01, 9.54915028e-02, 7.78360372e-02,
+            6.18466600e-02, 4.75864738e-02, 3.51117571e-02, 2.44717419e-02,
+            1.57084194e-02, 8.85637464e-03, 3.94264934e-03, 9.86635786e-04
+    };
+
+    template<typename R>
+    void test_hann()
+    {
+        constexpr R tol = std::is_same<R,float>::value ? 1e-3 : 1e-7;
```

Comment:
Same here and for any other `tol` values.  I bet these can all be a lot smaller :)